core/src/executor/evaluate/mod.rs

```diff
@@ -294,6 +301,81 @@ async fn evaluate_function<'a, 'b: 'a, 'c: 'a, T: GStore>(
             let exprs = stream::iter(exprs).then(eval).try_collect().await?;
             f::concat(exprs)
         }
+        Function::Custom { name, exprs } => {
+            let custom_func = storage
+                .fetch_function(name)
+                .await?
+                .ok_or_else(|| EvaluateError::UnsupportedFunction(name.to_string()))?;
+            let args: Vec<Evaluated<'_>> = stream::iter(exprs).then(eval).try_collect().await?;
+            let args: Vec<Value> = args
+                .into_iter()
+                .map(|v| Value::try_from(v).unwrap())
+                .collect();
+
+            let empty = vec![];
+
+            let fargs = custom_func.args.as_ref().unwrap_or(&empty);
+
+            let dargs: Vec<Value> = if let Some(fargs) = &custom_func.args {
+                let dargs: Vec<&Expr> =
+                    fargs.iter().filter_map(|y| y.default.as_ref()).collect();
+                let dargs: Vec<Evaluated<'_>> =
+                    stream::iter(dargs).then(eval).try_collect().await?;
+                dargs
+                    .into_iter()
+                    .map(|expr| Value::try_from(expr).unwrap())
+                    .collect()
```

Comment:
[nit-picking]

```suggestion
            let dargs = if let Some(fargs) = &custom_func.args {
                let dargs =
                    fargs.iter().filter_map(|y| y.default.as_ref()).collect::<Vec<_>>();
                let dargs =
                    stream::iter(dargs).then(eval).try_collect::<Vec<_>>().await?;
                dargs
                    .into_iter()
                    .map(|expr| Value::try_from(expr).unwrap())
                    .collect::<Vec<_>>()
```

When using collect, it is more convenient for future code modifications to pass Vec<_> as an argument instead of explicitly declaring the type of the variable being bound with let.